### PR TITLE
QnATelemetry improvements

### PR DESCRIPTION
**Pull request by lzc850612**
## Description
Add knowledgeBaseId into telemetry
serialize the questions array instead of appending them with comma

## Related Issue
#461 
#462 

## Testing Steps
Deploy the bot, talk to the bot and check app insight to see the telemetry properties and metrics

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the appropriate unit tests
- [ ] I have updated related documentation

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Middleware/Telemetry/TelemetryQnAMaker.cs [hunk 1]
```diff
@@ -78,15 +84,15 @@ public new async Task<QueryResult[]> GetAnswersAsync(ITurnContext context)
...
+                    telemetryProperties.Add(QnATelemetryConstants.QuestionProperty, JsonConvert.SerializeObject(queryResult.Questions));
                     telemetryProperties.Add(QnATelemetryConstants.AnswerProperty, queryResult.Answer);
                     telemetryMetrics.Add(QnATelemetryConstants.ScoreProperty, queryResult.Score);
                     telemetryProperties.Add(QnATelemetryConstants.ArticleFoundProperty, "true");
                 }
                 else
                 {
                     telemetryProperties.Add(QnATelemetryConstants.QuestionProperty, "No Qna Question matched");
-                    telemetryProperties.Add(QnATelemetryConstants.AnswerProperty, "No Qna Question matched");
+                    telemetryProperties.Add(QnATelemetryConstants.AnswerProperty, "No Qna Answer matched");
```
Comment (ryanisgrig) [inline solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Middleware/Telemetry/TelemetryQnAMaker.cs:95]: Good catch

**Comment by ryanisgrig** [comment]
@daveta FYI, for the next round of sample updates

**Review by ryanisgrig — APPROVED** [review]

**Review by ryanisgrig — CHANGES_REQUESTED** [review]
Can you port over/track the enterprise template updates as well?

**Comment by lzc850612** [comment]
Updated enterprise template as well as the enterprise bot sample

**Review by ryanisgrig — APPROVED** [review]

**Pull request merged**